From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: pthier <pthier@chromium.org>
Date: Fri, 21 Aug 2026 11:58:35 +0200
Subject: [M152] Native builtins are strict

Original change's description:
> Native builtins are strict
>
> There was a discrepancy between native builtins registered in
> bootstrapper vs setup-heap-internal.
> The former marked the builtins as strict, whereas the latter didn't.
>
> Mark builtins as strict also in setup-heap-internal.
>
> Fixed: 545767601
> Change-Id: Ida4da4bdcfe7dcfdc16f1da0ec022a65cf853ce8
> Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/8274474
> Reviewed-by: Leszek Swirski <leszeks@chromium.org>
> Commit-Queue: Patrick Thier <pthier@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#109425}

(cherry picked from commit 6f81b563fa9e3fde15412b0e3ab2b1e2b7e6d0d7)

Bug: 550760718,545767601
Change-Id: Ida4da4bdcfe7dcfdc16f1da0ec022a65cf853ce8
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/8279954
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/15.2@{#36}
Cr-Branched-From: a8fe17c242678787f03a8a2b77424aa06cf865c4-refs/heads/15.2.124@{#1}
Cr-Branched-From: 0e6ea09ef6f15ebd7f78555c0648c8b632ce3013-refs/heads/main@{#108894}

diff --git a/src/heap/setup-heap-internal.cc b/src/heap/setup-heap-internal.cc
index 7cabdd0516852d970395f3cddf02f082bdbd5701..7e3add2ab883b350663de38b85e4bacd10112827 100644
--- a/src/heap/setup-heap-internal.cc
+++ b/src/heap/setup-heap-internal.cc
@@ -74,6 +74,7 @@ DirectHandle<SharedFunctionInfo> CreateSharedFunctionInfo(
   DirectHandle<SharedFunctionInfo> shared =
       isolate->factory()->NewSharedFunctionInfoForBuiltin(
           isolate->factory()->empty_string(), builtin, len, kAdapt, kind);
+  shared->set_language_mode(LanguageMode::kStrict);
   return shared;
 }
 
